src/app/view/perspectives/chart/visualizer/plot-maker/plot-maker.ts

```diff
@@ -30,15 +33,65 @@ export abstract class PlotMaker {
 
   protected config: ChartConfig;
 
+  protected onValueChanged?: (valueChange: ValueChange) => void;
+
+  protected onDataChanged?: (dataChange: DataChange) => void;
```

Comment:
What is the difference between value and data change? Is the former change caused by user moving the point and the latter triggered by a new value in the store?

Reply:
DataChange is triggered while dragging and we update current shown data. DataChange is triggered after drag ends and we update data on the server. But I agree that names are not well choosen.

Reply:
The names are fine. I just wanted to make sure I understand how it works.

Reply:
Sorry, the second is is ValueChange